src/common/utils/e2e.ts

```diff
@@ -0,0 +1,71 @@
+import { SOCIAL_LOGIN_TYPE } from 'common/enums'
+import { isProd } from 'common/environment'
+import {
+  CodeExpiredError,
+  CodeInvalidError,
+  UnknownError,
+  CodeInactiveError,
+  OAuthTokenInvalidError,
+} from 'common/errors'
+
+const PASSPHREASE_EXPIRED = 'e2ets-loent-loent-loent-loent-expir'
+const PASSPHREASE_MISMATCH = 'e2ets-loent-loent-loent-loent-misma'
+const PASSPHREASE_UNKNOWN = 'e2ets-loent-loent-loent-loent-unkno'
+
+const CODE_NOT_EXIST = 'e2etest-code-not-exists'
+const CODE_RETIRED = 'e2etest-code-retired'
+const CODE_EXPIRED = 'e2etest-code-expired'
+
+export const checkIfE2ETest = (emailOrAuthCode: string) => {
+  if (isProd) {
+    return false
+  }
+  const isE2ETestEmail = /e2etest.*@matters.town/.test(emailOrAuthCode)
+  const isE2ETestOauthCode = /e2etest-.*/.test(emailOrAuthCode)
+
+  return isE2ETestEmail || isE2ETestOauthCode
+}
+
+export const throwIfE2EMagicToken = (token: string) => {
+  if (token === PASSPHREASE_EXPIRED) {
+    throw new CodeExpiredError('passphrases expired')
+  } else if (token === PASSPHREASE_MISMATCH) {
+    throw new CodeInvalidError('passphrases mismatch')
```

Comment:
changed to `PasswordInvalidError` as we [discussed](https://matterslab.slack.com/archives/C88CK7Q7L/p1694510415104729?thread_ts=1694420302.412109&cid=C88CK7Q7L)

Reply:
OK, I will rebase this pr base on your fix